packages/quantic/force-app/solutionExamples/main/lwc/exampleInsightPanel/exampleInsightPanel.js

```diff
@@ -124,4 +160,84 @@ export default class ExampleInsightPanel extends LightningElement {
       }
     );
   }
+
+  /**
+   * Indicates whether the watched fields of a case record have been updated.
+   * @param {object} newRecordData the new case record data
+   * @returns {boolean}
+   */
+  watchedFieldsUpdated(newRecordData) {
+    return Object.values(this.caseFields)?.reduce((result, field) => {
+      const previousValue = this.getFieldValueFromRecord(
+        this.caseRecord,
+        field
+      );
+      const newValue = this.getFieldValueFromRecord(newRecordData, field);
+      return result || previousValue !== newValue;
+    }, false);
+  }
+
+  /**
+   * Returns the value of a record field.
+   * This function expects the fieldName to be in the following format "Case.Contact.Name".
+   * @param {{fields: any}} record The case record
+   * @param {string} fieldName The case field name
+   * @returns {string}
+   */
+  getFieldValueFromRecord(record, fieldName) {
+    let innerKeys = fieldName.split('.');
+    if (!record || !innerKeys.length) return null;
+    if (innerKeys[0].toLowerCase() === 'case') {
+      innerKeys = innerKeys.slice(1);
+    }
+    if (innerKeys.length === 1) return record.fields?.[innerKeys[0]]?.value;
+    return this.getFieldValueFromRecord(
+      record.fields?.[innerKeys[0]]?.value,
+      innerKeys.slice(1).join('.')
+    );
+  }
+
+  /**
+   * Setup the case context.
+   * @returns {void}
+   */
+  setupContext() {
+    const context = {
+      Case_ID: this.caseId,
+      Case_Subject: this.getFieldValueFromRecord(
+        this.caseRecord,
+        'Case.Subject'
+      ),
+      Case_Description: this.getFieldValueFromRecord(
+        this.caseRecord,
+        'Case.Description'
+      ),
+    };
+    this.engine.dispatch(this.actions.setCaseContext(context));
+  }
+
+  /**
+   * Executes a search after detecting context change.
+   * @returns {void}
+   */
+  executeSearchAfterContextChanged() {
+    this.setupContext();
+    this.engine.dispatch(
+      this.actions.executeSearch(
+        this.actions.logContextChanged(this.caseId, this.caseNumber)
+      )
+    );
+  }
+
+  /**
+   * Executes the first search.
+   * @returns {void}
+   */
+  executeFirstSearch = () => {
```

Comment:
Curious why you'd need a separate function that actually does nothing if `triggeredFirstSearch` is true?

I think it would be better to simply either execute that code right before setting `triggeredFirstSearch = true`, or just removing that check and making sure you only call the function at the right moment.

Reply:
I removed the guard outside the method, let me know what you think